daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -89,11 +89,22 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
     *                      try blocks (interpretation) or Rollback nodes (iteration).
     */
   case class State private (
+      locallyCreated: Set[ContractId],
```

Comment:
Can we drop the `localContracts` Set from PartialTransaction so we don’t track this twice? I think a comment here would also be useful.